ios/engine/build.sh

```diff
@@ -0,0 +1,168 @@
+
+builder_describe "Builds Keyman Engine for use on iOS devices - iPhone and iPad." \
+  "@/web/src/app/webview        build" \
+  "@/common/web/sentry-manager  build" \
+  "clean" \
+  "configure" \
+  "build" \
+  "--debug         Avoids codesigning and adds full sourcemaps for the embedded predictive-text engine" \
+  "--sim-artifact  Also outputs a simulator-friendly test artifact corresponding to the build"
+
+builder_parse "$@"
+
+CONFIG="release"
+if builder_is_debug_build; then
+  CONFIG="debug"
+fi
+
+builder_describe_outputs \
+  configure     /ios/Carthage/Build     \
+  build         /ios/build/Build/Products/Release/KeymanEngine.xcframework
+
+# Base definitions (must be before do_clean call)
+DERIVED_DATA="$KEYMAN_ROOT/ios/build"
+BUILD_PATH=$DERIVED_DATA/Build/Products
+
+# Extended path definitions
+KMEI_RESOURCES=engine/KMEI/KeymanEngine/resources
+BUNDLE_PATH=$KMEI_RESOURCES/Keyman.bundle/Contents/Resources
+KMW_ROOT=../web
+
+# Needed before `configure` action
+DEFAULT_KBD_ID="sil_euro_latin"
+DEFAULT_LM_ID="nrc.en.mtnt"
+
+# Engine library build path
+KEYMAN_XCFRAMEWORK=$BUILD_PATH/$CONFIG/KeymanEngine.xcframework
+
+XCODEFLAGS="-quiet -configuration $CONFIG"
+XCODEFLAGS_EXT="$XCODEFLAGS -derivedDataPath $DERIVED_DATA -workspace ../keymanios.xcworkspace"
+
+CODE_SIGN=
+if builder_has_option --debug; then
+  CODE_SIGN="CODE_SIGN_IDENTITY= CODE_SIGNING_REQUIRED=NO ${DEV_TEAM:-} CODE_SIGN_ENTITLEMENTS= CODE_SIGNING_ALLOWED=NO"
+fi
+
+### START OF THE BUILD ###
+
+function do_clean () {
+  # Possible TODO:  can we clean the engine target without also cleaning the app target?
+  rm -rf $BUILD_PATH
+  rm -rf Carthage
+}
+
+function carthage_die() {
+  local msg="$1"
+
+  # Don't leave a trace of the failed folder; we'd have to rebuild stuff anyway.
+  # This way, a later re-run doesn't think `configure` succeeded when it did not.
+  rm -rf Carthage
+  builder_die "$1"
+}
+
+function do_carthage() {
+  # Carthage setup must be handled from the directory with the Cartfile / Cartfile.resolved, it seems.
+  pushd .. > /dev/null
+
+  carthage checkout || carthage_die "Carthage dependency checkout failed"
+
+  # --no-use-binaries: due to https://github.com/Carthage/Carthage/issues/3134,
+  # which affects the sentry-cocoa dependency.
+  carthage build --use-xcframeworks --no-use-binaries --platform iOS || carthage_die "Carthage dependency loading failed"
+
+  popd > /dev/null
+}
+
+function do_packages() {
+  if ! [ -d "$BUNDLE_PATH" ]; then
+      mkdir -p "$BUNDLE_PATH"
+  fi
+
+  local base_dir="$(pwd)"
+
+  downloadKeyboardPackage "$DEFAULT_KBD_ID"  "$base_dir/$BUNDLE_PATH/$DEFAULT_KBD_ID.kmp"
+  downloadModelPackage    "$DEFAULT_LM_ID"   "$base_dir/$BUNDLE_PATH/$DEFAULT_LM_ID.model.kmp"
+}
+
+function do_configure ( ) {
+  do_packages
+  do_carthage
+}
+
+# Manages KeymanEngine.bundle, which is included inside the :engine target.
+function update_bundle ( ) {
+  if ! [ -d "$BUNDLE_PATH" ]; then
+      mkdir -p "$BUNDLE_PATH"
+  fi
```

Comment:
```suggestion
  mkdir -p "$BUNDLE_PATH"
```

No need to test for existence when using `-p`